File: src/main/kotlin/de/huemotion/ConfigManager.java
package de.huemotion;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.Properties;

public class ConfigManager {

    private final String filePath = System.getProperty("user.dir") + File.separator + "Credentials.cfg";

    private Properties properties;

    public ConfigManager() {
        properties = new Properties();
        if ((new File(filePath)).exists()) {
            try {
                properties.load(new FileInputStream(filePath));
            } catch (Exception e) {
                e.printStackTrace();
            }
        } else {
            properties.put("SpeechToTextUser", "");
            properties.put("SpeechToTextPassword", "");
            properties.put("ToneAnalyzerUser", "");
            properties.put("ToneAnalyzerPassword", "");
            try {
                properties.store(new FileOutputStream(filePath), null);
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }

    public String getSpeechToTextUser() {
        return properties.getProperty("SpeechToTextUser");
    }

    public String getSpeechToTextPassword() {
        return properties.getProperty("SpeechToTextPassword");
    }

    public String getToneAnalyzerUser() {
        return properties.getProperty("ToneAnalyzerUser");
    }

    public String getToneAnalyzerPassword() {
        return properties.getProperty("ToneAnalyzerPassword");
    }

}

File: src/main/kotlin/de/huemotion/ConfigManager.kt
package de.huemotion

import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException
import java.util.Properties

object ConfigManager {

    private val filePath = System.getProperty("user.dir") + File.separator + "Credentials.cfg"

    private val properties: Properties = Properties()

    val speechToTextUser: String
        get() = properties.getProperty("SpeechToTextUser")

    val speechToTextPassword: String
        get() = properties.getProperty("SpeechToTextPassword")

    val toneAnalyzerUser: String
        get() = properties.getProperty("ToneAnalyzerUser")

    val toneAnalyzerPassword: String
        get() = properties.getProperty("ToneAnalyzerPassword")

    init {
        if (File(filePath).exists()) {
            try {
                properties.load(FileInputStream(filePath))
            } catch (e: Exception) {
                e.printStackTrace()
            }

        } else {
            properties["SpeechToTextUser"] = ""
            properties["SpeechToTextPassword"] = ""
            properties["ToneAnalyzerUser"] = ""
            properties["ToneAnalyzerPassword"] = ""
            try {
                properties.store(FileOutputStream(filePath), null)
            } catch (e: IOException) {
                e.printStackTrace()
            }

        }
    }

}